classes/class-p4-control-panel.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+use P4EN\Controllers\Ensapi_Controller as ENS_API;
+
+if ( ! class_exists( 'P4_Control_Panel' ) ) {
+
+	/**
+	 * Class P4_Control_Panel
+	 */
+	class P4_Control_Panel {
+
+		/**
+		 * P4_Control_Panel constructor.
+		 */
+		public function __construct() {
+			$this->hooks();
+		}
+
+		/**
+		 * Hooks actions and filters.
+		 */
+		public function hooks() {
+			if ( ! current_user_can( 'manage_options' ) ) {
+				return;
+			}
+			add_action( 'wp_dashboard_setup',           array( $this, 'add_dashboard_widgets' ), 9 );
+
+			add_action( 'wp_ajax_check_cache',          array( $this, 'check_cache' ) );
+			add_action( 'wp_ajax_check_en',             array( $this, 'check_en' ) );
+
+			add_action( 'admin_enqueue_scripts',        array( $this, 'enqueue_admin_assets' ) );
+		}
+
+		/**
+		 * .
+		 */
+		public function add_dashboard_widgets() {
+			wp_add_dashboard_widget(
+				'planet4_control_panel',
+				__( 'Planet 4 Control Panel', 'planet4-master-theme' ),
+				array( $this, 'add_items' )
+			);
+		}
+
+		/**
+		 * Outputs the contents of the dashboard widget.
+		 */
+		public function add_items() {
+			wp_nonce_field( 'cp-action' );
+
+			$this->add_item( [
+				'title'    => __( 'Cache', 'planet4-master-theme' ),
+				'subitems' => [
+					[
+						'title'  => __( 'Check Object Cache', 'planet4-master-theme' ),
+						'action' => 'check_cache',
+					],
+				],
+			] );
+
+			$this->add_item( [
+				'title'    => __( 'Engaging Networks', 'planet4-master-theme' ),
+				'subitems' => [
+					[
+						'title'  => __( 'Check Engaging Networks', 'planet4-master-theme' ),
+						'action' => 'check_en',
+					],
+				],
+			] );
+		}
+
+		/**
+		 * Adds a new item in the Control Panel and all of its subitems.
+		 *
+		 * @param array $data Associative array with all the data needed to add a new item in the Control Panel.
+		 */
+		public function add_item( $data ) {
+			echo '<div class="cp-item">
+					<div class="welcome-panel"><span><strong>' . esc_html( $data['title'] ) . '</strong></span>';
+			foreach ( $data['subitems'] as $subitem ) {
+				echo '<div>
+						<a href="#" class="btn btn-cp-action btn-' . esc_attr( $subitem['action'] ) . '-async" data-action="' . esc_attr( $subitem['action'] ) . '">' . esc_html( $subitem['title'] ) . '</a>
+						<span class="cp-subitem-response"></span>
+					</div>';
+			}
+			echo '</div>
+				</div>';
+		}
+
+		/**
+		 * Adds a check cache button to check connectivity to the Redis server.
+		 */
+		public function check_cache() {
+			if ( wp_doing_ajax() ) {
+				if ( ! current_user_can( 'manage_options' ) ) {
+					return;
+				}
+
+				if ( wp_verify_nonce( $_GET['_wpnonce'], 'cp-action' )  // CSRF protection.
```

Comment:
Is there any chance $_GET['_wpnonce'] would be unset, leading to a notice errror?